contracts/seed/Seed.sol

```diff
@@ -127,112 +131,124 @@ contract Seed {
+                                        - var to store the amount of seed to be distributed after calculation here
       * @param _price                 The price in wei of fundingTokens when exchanged for seedTokens.
       * @param _startTime             Distribution start time in unix timecode.
       * @param _endTime               Distribution end time in unix timecode.
-      * @param _vestingDuration       Vesting period duration in days.
-      * @param _vestingCliff          Cliff duration in days.
-      * @param _permissionedSeed         Set to true if only whitelisted adresses are allowed to participate.
+      * @param _vestingDuration       Vesting period duration in seconds.
+      * @param _vestingCliff          Cliff duration in seconds.
+      * @param _permissionedSeed      Set to true if only whitelisted adresses are allowed to participate.
       * @param _fee                   Success fee expressed as a % (e.g. 2 = 2% fee)
     */
     function initialize(
         address _beneficiary,
         address _admin,
         address[] memory _tokens,
-        uint256[] memory _softAndHardCap,
+        uint256[] memory _softHardReq,
         uint256 _price,
         uint256 _startTime,
         uint256 _endTime,
-        uint16  _vestingDuration,
-        uint16  _vestingCliff,
+        uint32  _vestingDuration,
+        uint32  _vestingCliff,
         bool    _permissionedSeed,
         uint8   _fee
     ) public initializer {
-        beneficiary     = _beneficiary;
-        admin           = _admin;
-        softCap         = _softAndHardCap[0];
-        hardCap         = _softAndHardCap[1];
-        price           = _price;
-        startTime       = _startTime;
-        endTime         = _endTime;
-        vestingDuration = _vestingDuration;
-        vestingCliff    = _vestingCliff;
-        permissionedSeed   = _permissionedSeed;
-        seedToken       = IERC20(_tokens[0]);
-        fundingToken    = IERC20(_tokens[1]);
-        fee             = _fee;
-        closed          = false;
-        minimumReached  = false;
-        maximumReached  = false;
-        seedRemainder   = IERC20(_tokens[0]).balanceOf(address(this));
+        beneficiary       = _beneficiary;
+        admin             = _admin;
+        softCap           = _softHardReq[0];
+        hardCap           = _softHardReq[1];
+        price             = _price;
+        startTime         = _startTime;
+        endTime           = _endTime;
+        vestingDuration   = _vestingDuration;
+        vestingCliff      = _vestingCliff;
+        permissionedSeed  = _permissionedSeed;
+        seedToken         = IERC20(_tokens[0]);
+        fundingToken      = IERC20(_tokens[1]);
+        fee               = _fee;
+        closed            = false;
+        minimumReached    = false;
+        maximumReached    = false;
+        seedRemainder     = _softHardReq[2];
+        seedAmountAtStart = _softHardReq[2];
```

Comment:
why `seedRemainder` is the hard cap? Hard cap should be denominated in funding tokens

Reply:
and why do we use a separate variable `seedAmountAtStart`?

Reply:
Okay, so the Index 2 is the seedToken amount transferred to the Seed when it was deployed, 
so at the beginning, seedRemainder is the total amount of seedToken sent.
Doug wanted a variable to keep track of the amount of seed allocated at the time it was deployed.

Reply:
Added to Issue #160 , This PR is getting very big. Closing this request.